cos-gpu-installer-docker/entrypoint.sh

```diff
@@ -181,9 +189,32 @@ download_kernel_src() {
   popd
 }
 
-install_cross_toolchain_pkg() {
-  mkdir -p /build
-  pushd /build
+# Download content from a given URL to specific location.
+#
+# Args:
+# download_url: The URL used to download the archive/file.
+# output_name: Output name of the downloaded archive/file.
+# info_str: Describes the archive/file that is downloaded.
+download_content_from_url() {
+  local -r download_url=$1
+  local -r output_name=$2
+  local -r info_str=$3
+
+  local attempts=0
+  until time curl -sfS "${download_url}" -o "${output_name}"; do
+    attempts=$(( ${attempts} + 1 ))
+    if (( "${attempts}" >= "${RETRY_COUNT}" )); then
+      error "Could not download ${info_str} from ${download_url}, giving up."
+	  break
```

Comment:
This function should `return 1` if it couldn't download from the url.

Then line 243 and line 266 can be just like
`if ! download_content_from_url "${TOOLCHAIN_DOWNLOAD_URL}" "${pkg_name}" "toolchain archive"; then
   return ${RETCODE_ERROR}
fi`